### PR TITLE
Us12 merchant items index 5 most popular items

### PR DESCRIPTION
**Pull request by thawley2**
- Added enabled or disabled status test in item model spec. 
- Added validations to items in order to test sad path for item update action in items_controller. 
- Added test for #full_name in the customer model. 
- Created new test data in a separate method called merchant2_test_data to test the method that was created called 
    top_five_items in the merchant model. 
- Added a query in the merchant model for top_five_items with tests in the model spec along with the merchant/items/index spec. 
- Updated the merchant/items/index view to include div tags to better test the returned data. 
- Updated the _enable_disable partial to include more repeated code. 
- Created new partial called _merchant_header that includes "Little Esty Shop and merchant name"
- Updated all Merchant pages withe the new _merchant_header partial
- Refactored #items_ready_to_ship method test in the merchant model spec to not include the date formatting since we are formatting the date in the view.

### CODE REVIEW

**Comment by thawley2** [comment]
Looks like there was a little merge conflict since we were adding stuff to the same 3 files.  I kept everything, so nothing should be deleted, just shifted up or down.

**Review by amspears007 — APPROVED** [review]
Awesome!

File: app/views/merchants/items/edit.html.erb [hunk 1]
```diff
@@ -1,5 +1,3 @@
-<h1>Little Esty Shop</h1>
-
-<h2><%= @merchant.name %></h2>
+<%= render partial: '/merchants/header', locals: {merchant: @merchant.name} %>
```
Comment (sicknacco) [inline app/views/merchants/items/edit.html.erb:1]: Ooooo. Good idea!

File: app/views/merchants/items/index.html.erb [hunk 2]
```diff
@@ -1,29 +1,42 @@
...
-<ul style="list-style-type:none">
   <% @merchant.enabled_items.each do |item| %>
-  <div id="item_<%=item.id %>">
-    <p><li><%= link_to item.name, merchant_item_path(@merchant, item) %> </li></p>
-    <%= render partial: 'enable_disable', locals: {model: [@merchant, item], value: 'disabled', button_text: 'Disable'} %>
-  </div>
+    <%= render partial: 'enable_disable', 
+      locals: {item_id: item.id, 
+      item_name: item.name, 
+      model: [@merchant, item],
```
Comment (sicknacco) [inline app/views/merchants/items/index.html.erb:12]: This formatting is great.

File: app/views/merchants/items/index.html.erb [hunk 3]
```diff
@@ -1,29 +1,42 @@
...
+<h3>Disabled Items</h3>
   <% @merchant.disabled_items.each do |item| %>
-  <div id="item_<%=item.id %>">
-    <p><li><%= link_to item.name, merchant_item_path(@merchant, item) %></li></p>
-    <%= render partial: 'enable_disable', locals: {model: [@merchant, item], value: 'enabled', button_text: 'Enable'} %>
-  </div>
+    <%= render partial: 'enable_disable', 
+      locals: {item_id: item.id, 
+      item_name: item.name, 
+      model: [@merchant, item],
```
Comment (sicknacco) [inline app/views/merchants/items/index.html.erb:25]: This is much more legible! I wouldn't have thought of this

**Review by sicknacco — APPROVED** [review]
Awesome refactors and great ActiveRecord!

**Pull request merged**